### PR TITLE
addition of schemaVersion check and update index

### PR DESCRIPTION
**Pull request by jh-bate**
@darinkrauss here is the change for using the schema version

### CODE REVIEW

File: config/server.json [hunk 1]
```diff
@@ -8,5 +8,6 @@
   },
   "mongo": {
     "connectionString": "mongodb://localhost/streams"
-  }
+  },
+  "schemaVersionGte": 1
```
Comment (darinkrauss) [inline config/server.json]: What does the "Gte" mean?

File: tide-whisperer.go [hunk 2]
```diff
@@ -27,8 +27,9 @@ import (
 type (
 	Config struct {
 		clients.Config
-		Service disc.ServiceListing `json:"service"`
-		Mongo   mongo.Config        `json:"mongo"`
+		Service          disc.ServiceListing `json:"service"`
+		Mongo            mongo.Config        `json:"mongo"`
+		AllowedSchemaGte int                 `json:"schemaVersionGte"`
```
Comment (darinkrauss) [inline tide-whisperer.go]: For consistency's sake, I would use "SchemaVersionGte" for the field name.

File: config/server.json (hunk 1 above)
Comment (darinkrauss) [inline config/server.json]: I assume this means greater-than-or-equal-to, yes?  Is this for pre-UTC or post-UTC deploy?  We may need to have a schema version min and max.  Probably should discuss.

File: tide-whisperer.go [hunk 3]
```diff
@@ -224,7 +225,7 @@ func main() {
 		defer mongoSession.Close()
 
 		//select this data
-		groupDataQuery := bson.M{"_groupId": groupId, "_active": true}
+		groupDataQuery := bson.M{"_groupId": groupId, "_active": true, "_schemaVersion": bson.M{"$gte": config.AllowedSchemaGte}}
```
Comment (darinkrauss) [inline tide-whisperer.go]: A schema range (min, max) would change this query.  See above.

**Comment by jh-bate** [comment]
@darinkrauss I have updated the PR to set an upper and lower schema version

File: config/server.json [hunk 4]
```diff
@@ -8,5 +8,9 @@
   },
   "mongo": {
     "connectionString": "mongodb://localhost/streams"
+  },
+  "schemaVersion": {
+    "gte": 0,
```
Comment (darinkrauss) [inline config/server.json]: Might be clearer if it was "minimum" and "maximum" (rather than "gte" and "lte"). Just my 2 cents, though...

**Comment by darinkrauss** [comment]
One comment, otherwise :ship:

File: config/server.json (hunk 4 above)
Comment (jh-bate) [inline config/server.json]: yeah true - I was mapping it to the mongo query :) I will change that and then merge

**Pull request merged**